core/blockchain.go

```diff
@@ -2527,11 +2531,16 @@ func (bc *BlockChain) ValidatePayload(block *types.Block, feeRecipient common.Ad
 	// and dangling prefetcher, without defering each and holding on live refs.
 	defer statedb.StopPrefetcher()
 
+	feeRecipientBalanceBefore := statedb.GetBalance(feeRecipient)
```

Comment:
Underneath this function does the following:
```
func (s *stateObject) Balance() *big.Int {
	return s.data.Balance
}
```

and none of the data is copied, so here you hold the actual balance pointer of the object. I'd propose to copy it out of precaution so that people using the variable don't mistakenly change the actual balance in statedb, like you do with feeRecipientBalanceDelta. I think the optimization here is too risky

Reply:
sure, why not

btw we use the same pattern for bundle profit in our builder

today it works if you don't change this variable because in geth they chose to return pointer here but when balance is changed new value is created

but who knows what change can happen later